pkg/daemon/update.go

```diff
@@ -579,29 +579,29 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		glog.Info("Changes do not require drain, skipping.")
 	}
 
+	oldIgnConfig, err := ctrlcommon.ParseAndConvertConfig(oldConfig.Spec.Config.Raw)
+	if err != nil {
+		return fmt.Errorf("parsing old Ignition config failed with error: %v", err)
+	}
+	newIgnConfig, err := ctrlcommon.ParseAndConvertConfig(newConfig.Spec.Config.Raw)
+	if err != nil {
+		return fmt.Errorf("parsing new Ignition config failed with error: %v", err)
+	}
+
 	// update files on disk that need updating
-	if err := dn.updateFiles(oldConfig, newConfig); err != nil {
+	if err := dn.updateFiles(&oldIgnConfig, &newIgnConfig); err != nil {
```

Comment:
Since we are not going to modify params, let's pass them by value.

Reply:
`updateFiles` calls `deleteStaleData` which takes references, but I could just pass references to the copy.

In general, when is it good practice to pass by value? Is it just a tradeoff between safety/no garbage collection vs size of a copy?